src/routes/sitemaps.ts

```diff
@@ -25,4 +34,49 @@ export default async function (fastify: FastifyInstance): Promise<void> {
       .header('cache-control', 'public, max-age=14400')
       .send(stream);
   });
+
+  fastify.get('/robots.txt', async (req, res) => {
+    const con = await createOrGetConnection();
+    const bots = con
+      .createQueryBuilder()
+      .select('username')
+      .from(BotUser, 'b')
+      .orderBy('"createdAt"', 'DESC')
+      .limit(10_000)
+      .getQuery();
+
+    const freshUsers = con
+      .createQueryBuilder()
+      .select('username')
+      .from(User, 'u')
+      .orderBy('"createdAt"', 'DESC')
+      .where('"createdAt" > now() - interval \'14 day\'')
+      .andWhere('reputation <= 10')
+      .limit(10_000)
+      .getQuery();
+
+    const queryStream = await con
+      .createQueryBuilder()
+      .select('*')
+      .from(`((${bots}) UNION (${freshUsers}))`, 'a')
+      .stream();
+
+    const usersStream = queryStream.map(
+      (row: Pick<User, 'username'>) => `Disallow: /${row.username}\n`,
+    );
+
+    const baseRobotsStream = new Readable();
+    baseRobotsStream._read = () => {};
+    baseRobotsStream.push(BASE_ROBOTS);
+    baseRobotsStream.push(null);
+
+    let stream = new PassThrough();
+    stream = baseRobotsStream.pipe(stream, { end: false });
+    stream = usersStream.pipe(stream, { end: true });
+
+    return res
+      .type('text/plain')
+      .header('cache-control', 'public, max-age=3600')
```

Comment:
Does this actually caches on the CDN? We don't have one for API, thinking since it will be rewrote in vercel we could add s-maxage so their CDN picks it up and caches for a while?

Reply:
We do have CDN enabled for the API side. Can add also s-maxage if needed, no problem

Reply:
Nice 👌

Reply:
Yeah, s-maxage is mostly used for server side expiration so I would add it if CDN knows to pick it up.

Reply:
Added it everywhere